code/game/objects/items/RPD.dm

```diff
@@ -139,14 +139,13 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 			dirs = list("[NORTH]" = "North", "[EAST]" = "East", "[SOUTH]" = "South", "[WEST]" = "West",
 						"[NORTHEAST]" = "North Flipped", "[SOUTHEAST]" = "East Flipped", "[SOUTHWEST]" = "South Flipped", "[NORTHWEST]" = "West Flipped")
 
-
 	var/list/rows = list()
 	var/list/row = list("previews" = list())
 	var/i = 0
 	for(var/dir in dirs)
 		var/numdir = text2num(dir)
 		var/flipped = ((dirtype == PIPE_TRIN_M) || (dirtype == PIPE_UNARY_FLIPPABLE)) && (ISDIAGONALDIR(numdir))
-		row["previews"] += list(list("selected" = (numdir == selected_dir), "dir" = dir2text(numdir), "dir_name" = dirs[dir], "icon_state" = icon_state, "flipped" = flipped))
+		row["previews"] += list(list("selected" = dirtype == PIPE_ONEDIR ? TRUE : (numdir == selected_dir), "dir" = dir2text(numdir), "dir_name" = dirs[dir], "icon_state" = icon_state, "flipped" = flipped))
```

Comment:
```suggestion
		row["previews"] += list(list(
			"selected" = dirtype == PIPE_ONEDIR ? TRUE : (numdir == selected_dir),
			"dir" = dir2text(numdir), 
			"dir_name" = dirs[dir],
			"icon_state" = icon_state,
			"flipped" = flipped,
		))
```

much easier to read. if this will work alphabetized, pass that in too.